nupic/research/frameworks/pytorch/imagenet/parser_utils.py

```diff
@@ -0,0 +1,165 @@
+
+from nupic.research.frameworks.dynamic_sparse.common.ray_custom_loggers_2 import (
+    DEFAULT_LOGGERS,
+)
+from nupic.research.frameworks.pytorch.imagenet import mixins
+from nupic.research.frameworks.sigopt.sigopt_experiment import SigOptImagenetExperiment
+from nupic.research.frameworks.wandb import ray_wandb
+
+MAIN_PARSER = argparse.ArgumentParser(
+    formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    argument_default=argparse.SUPPRESS,
+    add_help=False,
+)
+MAIN_PARSER.add_argument("-g", "--num-gpus", type=int,
+                         default=torch.cuda.device_count(),
+                         help="number of GPUs to use")
+MAIN_PARSER.add_argument("-n", "--num-cpus", type=int,
+                         default=torch.get_num_interop_threads(),
+                         help="number of CPUs to use when GPU is not available."),
+MAIN_PARSER.add_argument("-r", "--restore", action="store_true",
+                         help="Restore training from last known checkpoint")
+MAIN_PARSER.add_argument("-c", "--checkpoint-file", dest="restore_checkpoint_file",
+                         help="Resume experiment from specific checkpoint file")
+MAIN_PARSER.add_argument("-d", "--copy-checkpoint-to-dir", dest="copy_checkpoint_dir",
+                         help="Copy final saved checkpoint to specified directory.")
+MAIN_PARSER.add_argument("-j", "--workers", type=int, default=4,
+                         help="Number of dataloaders workers")
+MAIN_PARSER.add_argument("-b", "--backend", choices=["nccl", "gloo"],
+                         help="Pytorch Distributed backend", default="nccl")
+MAIN_PARSER.add_argument("-p", "--progress", action="store_true",
+                         help="Show progress during training")
+MAIN_PARSER.add_argument("-l", "--log-level",
+                         choices=["critical", "error", "warning", "info", "debug"],
+                         help="Python Logging level")
+MAIN_PARSER.add_argument("-f", "--log-format",
+                         help="Python Logging Format")
+MAIN_PARSER.add_argument("-x", "--max-failures", type=int,
+                         help="How many times to try to recover before stopping")
+MAIN_PARSER.add_argument("--checkpoint-freq", type=int,
+                         help="How often to checkpoint (epochs)")
+MAIN_PARSER.add_argument("--profile", action="store_true",
+                         help="Enable cProfile tracing")
+MAIN_PARSER.add_argument("--profile-autograd", action="store_true",
+                         help="Enable torch.autograd.profiler.profile during training")
+MAIN_PARSER.add_argument("--wandb", action="store_true",
+                         help="Enable logging through wandb.")
+MAIN_PARSER.add_argument("--wandb_resume", action="store_true",
+                         help="Resume logging through wandb.")
+MAIN_PARSER.add_argument("-t", "--create_sigopt", action="store_true",
+                         help="Create a new sigopt experiment using the config")
```

Comment:
Maybe you could separate `wandb`, `sigopt` and any other 3rd party library options into different parsers, just like what you did for `RAY_PARSER` and keep the `MAIN_PARSE` clean from 3rd party dependencies

Reply:
Done. I also introduced a `DEFAULT_PARSERS` list with all of them. Let me know if you think this is reasonable (it looks cleaner to me to have to import only one thing as opposed to four individual parsers when possible).